code/modules/cm_marines/equipment/sentries.dm

```diff
@@ -1092,10 +1262,197 @@
+	health_max = 150
+	rounds = 500
+	rounds_max = 500
+	knockdown_threshold = 70 //lighter, not as well secured.
+	work_time = 20 //significantly faster than the big sentry
+	ammo = /datum/ammo/bullet/turret/mini //Similar to M39 AP rounds.
+	magazine_type = /obj/item/ammo_magazine/minisentry
+
+/obj/item/storage/box/sentry/New()
+	..()
+	update_icon()
+
+/obj/machinery/marine_turret/mini/MouseDrop(over_object, src_location, over_location) //Drag the tripod onto you to fold it.
+	if(!ishuman(usr))
+		return
+	var/mob/living/carbon/human/user = usr //this is us
+	if(over_object == user && in_range(src, user))
+		if(anchored)
+			to_chat(user, "<span class='warning'>You must unanchor [src] to retrieve it!</span>")
+			return
+		if(on)
+			to_chat(user, "<span class='warning'>You depower [src] to facilitate its retrieval.</span>")
+			on = FALSE
+			update_icon()
+
+		user.visible_message("<span class='notice'>[user] begins to fold up and retrieve [src].</span>",
+		"<span class='notice'>You begin to fold up and retrieve [src].</span>")
+		if(do_after(user, work_time * 2, TRUE, 5, BUSY_ICON_BUILD))
+			if(!src || on || anchored)//Check if we got exploded
+				return
+			to_chat(user, "<span class='notice'>You fold up and retrieve [src].</span>")
+			var/obj/item/device/marine_turret/mini/P = new(loc)
+			user.put_in_hands(P)
+			P.health = health //track the health
+			cdel(src)
+
+/obj/machinery/marine_turret/mini/update_icon()
+	if(stat && health > 0) //Knocked over
+		on = FALSE
+		density = FALSE
+		icon_state = "minisentry_fallen"
+		return
+	else
+		density = initial(density)
+
+	if(!cell)
+		on = FALSE
+		icon_state = "minisentry_nobat"
+		return
+
+	if(cell.charge <= 0)
+		on = FALSE
+		icon_state = "minisentry_nobat"
+		return
+
+	if(!rounds)
+		icon_state = "minisentry_noammo"
+		return
+
+	if(on)
+		start_processing()
+		if(!radial_mode)
+			icon_state = "minisentry_on"
+		else
+			icon_state = "minisentry_on_radial"
+	else
+		icon_state = "minisentry_off"
+		stop_processing()
+
+
+/obj/item/device/marine_turret/mini
+	name = "\improper UA-580 Point Defense Sentry (Folded)"
+	desc = "A deployable, automated turret with AI targeting capabilities. This is a lightweight portable model meant for rapid deployment and point defense. Armed with an light, high velocity machine gun and a 500-round drum magazine. It is currently folded up."
+	icon = 'icons/Marine/miniturret.dmi'
+	icon_state = "minisentry_packed"
+	item_state = "minisentry_packed"
+	w_class = 4
+	health = 150 //We keep track of this when folding up the sentry.
+	flags_equip_slot = SLOT_BACK
+
+/obj/item/device/marine_turret/mini/attack_self(mob/user) //click the sentry to deploy it.
+	if(!ishuman(usr))
+		return
+	var/turf/target = get_step(user.loc,user.dir)
+	if(!target)
+		return
+	var/blocked
+	if(target.density)
+		blocked = TRUE
+	if(!blocked) //allows us to stop checks and thus save resources if we find something that blocks placement
+		if(locate(/obj/machinery) in target)
```

Comment:
var/blocked = TRUE
	if(!target.density)
		for(var/obj/O in target)
			if(O.density && !(S.flags_atom & ON_BORDER))
				blocked = TRUE
				break
	else
		blocked = TRUE

Reply:
whoops, I didn't mean to default the var to true, fixed it for ya

Reply:
I was deliberately cutting down on categories of objects to the two that can actually block LoF to save resources

Reply:
I'd have to check if `locate(type) in list` saves resources before a `for(type in list)`